xonsh/dirstack.py

```diff
@@ -10,6 +10,75 @@
 """
 
 
+def _get_cwd():
+    try:
+        return os.getcwd()
+    except OSError:
+        return None
+    except FileNotFoundError:
+        return None
+
+
+def _change_working_directory(newdir):
+    env = builtins.__xonsh_env__
+    old = _get_cwd()
+    try:
+        os.chdir(newdir)
+    except:
```

Comment:
Just pushed this change (though I think it should try to catch `FileNotFoundError` as well, so I put that in).